mlos_bench/mlos_bench/config/optimizers/mlos_core_skopt.jsonc

```diff
@@ -1,5 +1,6 @@
 // MLOS Core optimizer
 {
+    "$schema": "../schemas/optimizers/optimizer-schema.json",
```

Comment:
Mostly agree.  I commented on some of this above already.  Here's some additional thoughts:

1. Yes, we want to use the schemas to validate the configs, but not the schema that's listed in the file.  In fact, we almost entirely ignore that field.  It purely is for the editor to use.
If you look in `ConfigSchema` we use the local schema files relative to the code always and remove the `$schema` attribute from the config (if present - we don't require it) once the schema validation has occurred, since as you rightly pointed out - we already know the expected config format in a given situation.

However, I will note that there's value in using the schema validator beyond just our code.  One example are extra attributes that are included in the config that aren't recognized or looked for in the code (e.g. due to misspellings or other reasons).  Especially when we pass those on to other code that's expecting a certain set of config k-v pairs this causes problems.

On the other hand, there are things that the schema can't enforce directly.  So, I guess what I'm arguing for is a belts *and* suspenders approach here: we want schema validation *and* code validation.

2. Yes, we can use "just" URLs (e.g. the rawcontent.github.com ones I documented in the `schemas/README.md` and the `$id` in the schema files themselves).
However, those aren't published yet (waiting on these PRs), so there were some editor issues.  The editor (or other validator) will need to do fetching of those (unless told otherwise like we do in our `.vscode/settings.json`), but the validator should cache them then after that.

Anyways, TL;DR: the local paths were helpful for me to get started with this, but I think we can switch over to URLs now.  I'll give it a shot and see how it goes.

Reply:
Unfortunately, the url -> local schema file redirect syntax doesn't support relative paths yet, so I think we need to just omit the schema ids for now.

https://github.com/Microsoft/vscode/issues/2809#issuecomment-1544387883

Would be nice to include some examples that do include those URLs for uses to reference, but maybe we kick that for future PR.  Thoughts?